mock-identity-system/src/test/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImplTest.java

```diff
@@ -118,4 +120,84 @@ public void kycAuth_withInValidKbaChallenge_thenFail() {
+        ReflectionTestUtils.setField(authenticationService, "fieldDetailList", fieldDetailList);
+        ReflectionTestUtils.setField(authenticationService, "fieldLang", "eng");
+        ReflectionTestUtils.setField(authenticationService,"objectMapper",new ObjectMapper());
+
+        KycAuthRequestDto kycAuthRequestDto = new KycAuthRequestDto();
+        kycAuthRequestDto.setKba("eyJmdWxsTmFtZSI6IlNpZGRoYXJ0aCBLIE1hbnNvdXIiLCJkYXRlT2ZCaXJ0aCI6IjE5ODctMTEtMjUifQ==");
+        kycAuthRequestDto.setIndividualId("individualId");
+        kycAuthRequestDto.setTransactionId("transactionId");
+
+        IdentityData identityData = new IdentityData();
+        identityData.setDateOfBirth("1987/11/25");
+        LanguageValue languageValueFullName = new LanguageValue();
+        languageValueFullName.setLanguage("eng");
+        languageValueFullName.setValue("Siddharth K Mansour");
+        identityData.setFullName(List.of(languageValueFullName));
+
+        LanguageValue languageValueName = new LanguageValue();
+        languageValueName.setLanguage("eng");
+        languageValueName.setValue("Siddharth");
+        identityData.setName(List.of(languageValueName));
+
+        identityData.setEncodedPhoto("endcodedPhoto");
+        identityData.setDateOfBirth("1987/11/25");
+
+        Mockito.when(identityService.getIdentity(Mockito.anyString())).thenReturn(identityData);
+
+        Mockito.when(authRepository.save(Mockito.any())).thenReturn(new KycAuth());
+
+        List<VerifiedClaim> verifiedClaimList =new ArrayList<>();
+        //fill this verifiedClaimList with some data
+        VerifiedClaim verifiedClaimName = new VerifiedClaim();
+        verifiedClaimName.setTrustFramework("trustFramework");
+        verifiedClaimName.setClaim("name");
+        verifiedClaimName.setIndividualId("individualId");
+        verifiedClaimName.setActive(true);
+        verifiedClaimName.setVerifiedDateTime(LocalDateTime.now());
+
+        VerifiedClaim verifiedClaimEmail = new VerifiedClaim();
+        verifiedClaimEmail.setTrustFramework("trustFramework");
+        verifiedClaimEmail.setClaim("email");
+        verifiedClaimEmail.setIndividualId("individualId");
+        verifiedClaimEmail.setActive(true);
+        verifiedClaimEmail.setVerifiedDateTime(LocalDateTime.now());
+
+        verifiedClaimList.add(verifiedClaimName);
+        verifiedClaimList.add(verifiedClaimEmail);
+
+
+        Optional<List<VerifiedClaim>> verifiedClaimsOptional = Optional.of(verifiedClaimList);
+        Mockito.when(verifiedClaimRepository.findByIndividualIdAndActive(Mockito.anyString(),Mockito.anyBoolean())).thenReturn(verifiedClaimsOptional);
+
+        KycAuthResponseDtoV2 kycAuthResponseDtoV2 = authenticationService.kycAuthV2("relyingPartyId", "clientId", kycAuthRequestDto);
+        List<ClaimMetadata> claimMetadataList = kycAuthResponseDtoV2.getClaimMetadataList();
+        for(ClaimMetadata claimMetadata : claimMetadataList){
```

Comment:
Here checking the size of claimMetadata and checking if only email and name is with verification data will be the correct assertion.

Reply:
done